src/main/java/com/fasterxml/jackson/databind/ser/BeanSerializer.java

```diff
@@ -147,7 +147,6 @@ protected BeanSerializerBase asArraySerializer()
          * - have per-property filters
          */
         if ((_objectIdWriter == null)
-                && (_anyGetterWriter == null)
```

Comment:
Do we know this (serialize-as-array) actually works? Is there a test for that?

Reply:
Some surprising updates,

1# While trying to add coverage for `As.Array` + `@JsonAnyGetter` via #4961 , found out the two never worked together
2# Merging #4961 to this branch, the output **"actually became better"** check [Github action I triggered.](https://github.com/FasterXML/jackson-databind/actions/runs/13216961122/job/36897304956)
3# I added comment in [below test](https://github.com/FasterXML/jackson-databind/pull/4775/files#r1947846664) from #4961

Honestly I don't clearly remember what I did/thought back then, but seems like additional improvement

Reply:
But still, let's see if we can actually flatten out as-array?

Reply:
Nvm, let's do separate PR for the flattening part.

Reply:
Ok. So, flattening is one part, but the other is that with Arrays, name is implied by position. This is why I am not sure any-getter can actually be supported with as-array Shape.
But what to do in that case: error out, or refuse to change output shape?

Reply:
Oh! I see -- instead of flattening entries of any-getter, they could instead be serialized as single Object with key/value pairs retained. That would actually make some sense since it would reliably map into specific index and retain information. It would not be really "as-array", but does best that can be achieved, I think.

I like that behavior, if I understood it correctly.

Reply:
The flattening out part suggested because I thought why we can do is serialize...

- First prop
- Second prop
- Any getter (information retained until serialized)
- Fourth prop
... something like that.

Reply:
But what you said about retaining index based property location-implication makes sense 👍🏼